images/dnsmasq/Makefile

```diff
@@ -89,74 +82,56 @@ all-containers: $(addprefix containers-, $(ALL_ARCH))
 all-test: $(addprefix test-, $(ALL_ARCH))
 
 .PHONY: all-push
-all-push: $(addprefix push-, $(ALL_ARCH))
+all-push: ./manifest-tool $(addprefix push-, $(ALL_ARCH))
+	./manifest-tool push from-args --platforms $(ML_PLATFORMS) --template $(REGISTRY)/$(IMAGENAME)-ARCH:$(VERSION) --target $(REGISTRY)/$(IMAGENAME):$(VERSION)
 
 .PHONY: build
 build: $(BINARY)
 
-$(BINARY): Dockerfile.cross $(DNSMASQ_ARCHIVE) dnsmasq.conf
+$(BINARY):
 	@echo "building :" $(BINARY)
 	@mkdir -p $(@D)
-	@cp Dockerfile.cross $(OUTPUT_DIR)/Dockerfile
-	@cd $(OUTPUT_DIR) && sed -i "s|__BASEIMAGE__|$(BASEIMAGE)|g" Dockerfile
-
-ifeq ($(ARCH),amd64)
-	@cd $(OUTPUT_DIR) && sed -i "/__CROSS_BUILD_COPY__/d" Dockerfile
-	@docker run --rm --sig-proxy=true     \
-		-v `pwd`/$(OUTPUT_DIR):/build \
-		-v `pwd`:/src                 \
-		$(COMPILE_IMAGE)              \
-		/bin/sh -c                    \
-		"apk update                                 \
-			&& apk add alpine-sdk xz libcap-dev \
-			&& tar -xJf /build/dnsmasq.tar.xz   \
-			&& cd $(DNSMASQ_VERSION)            \
-			&& make -j                          \
-			&& cp src/dnsmasq /build" $(VERBOSE_OUTPUT)
-else
-	@cd $(OUTPUT_DIR)                                            \
-		&& sed -i "s|__ARCH__|$(QEMUARCH)|g" Dockerfile      \
-		&& sed -i "s/__CROSS_BUILD_COPY__/COPY/g" Dockerfile
-	@cd $(OUTPUT_DIR) \
-		&& curl -sSL $(MULTIARCH_RELEASE) | tar -xJ
-	@docker run --sig-proxy=true --rm            \
-		--privileged $(MULTIARCH_CONTAINER)  \
-		--reset $(VERBOSE_OUTPUT)
-	@docker run --rm --sig-proxy=true            \
+	@docker run --rm --sig-proxy=true        \
 		-v `pwd`/$(OUTPUT_DIR):/build        \
-		-v `pwd`:/src                        \
 		$(COMPILE_IMAGE)                     \
 		/bin/bash -c                         \
-		"tar -xJf /build/dnsmasq.tar.xz      \
+		"curl -sSL $(DNSMASQ_URL) | tar -xJ  \
 			&& cd $(DNSMASQ_VERSION)     \
-			&& CC=$(TRIPLE)-gcc make -j  \
+			&& CC=$(CC) make -j  \
 			&& cp src/dnsmasq /build" $(VERBOSE_OUTPUT)
-endif
-	@cp dnsmasq.conf $(OUTPUT_DIR)
-
-$(DNSMASQ_ARCHIVE):
-	@mkdir -p $(@D)
-	@curl -sSL $(DNSMASQ_URL) > $@
 
 .PHONY: containers
 containers: $(CONTAINER_STAMP)
 
-$(CONTAINER_STAMP): $(BINARY)
-	@echo "container:" $(REGISTRY)/$(IMAGE):$(VERSION)
+$(CONTAINER_STAMP): Dockerfile dnsmasq.conf $(BINARY)
+	@echo "container:" $(REGISTRY)/$(IMAGENAME)-$(ARCH):$(VERSION)
+	@cp dnsmasq.conf $(OUTPUT_DIR)/
+	@sed "s|__BASEIMAGE__|$(BASEIMAGE)|g" Dockerfile > $(OUTPUT_DIR)/Dockerfile
+	@touch $(OUTPUT_DIR)/.emptydir
 	@docker build \
-		-q -t $(REGISTRY)/$(IMAGE):$(VERSION) $(OUTPUT_DIR) > $@
+		-q -t $(REGISTRY)/$(IMAGENAME)-$(ARCH):$(VERSION) $(OUTPUT_DIR) > $@
 
 .PHONY: test
 test: containers
-	@ARCH=$(ARCH) IMAGE=$(REGISTRY)/$(IMAGE):$(VERSION) ./validate.sh
+	@ARCH=$(ARCH) IMAGE=$(REGISTRY)/$(IMAGENAME)-$(ARCH):$(VERSION) ./validate.sh
+
+./manifest-tool:
+	curl -sSL https://github.com/luxas/manifest-tool/releases/download/v0.3.0/manifest-tool > manifest-tool
```

Comment:
What does manifest-tool do? Is it essential for the build process. I would prefer to minimize external dependencies (e.g. things that need to be downloaded) for the build.